content/docs/getting-started/hello-features.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Tokio Features"
+weight : 1020
+menu:
+  docs:
+    parent: getting_started
+---
+
+Feel free to skip this section, and just use `features = ["full"]` while
+learning. This section provides some examples of how and when to specify
+tokio features. 
+
+## Often using all features is fine 
+
+When writing an app, we'll often use all of Tokio's features to accelerate
+our development time:
+
+```toml
+[dependencies]
+tokio = { version = "0.2", features = ["full"] }
+```
+
+For most apps, the additional compile time is hardly noticeable relative to the
+cognitive overhead of thinking about which features are needed.
+
+# Selecting features
+
+When we develop protocol implementations that are used by client apps and
+servers, we want to limit the dependencies to decrease compile time --
+not just for us, but also for the users of our crates.
+
+Let's take a look at our tiny client app and see what features are needed.
+
+## tokio main
+
+If we change our code, so that it only uses the `#[tokio::main]` macro:
+
+```
```

Comment:
Since we're essentially rewriting the hello world example, it might be nice to include the entire file (since it's just a couple of lines more).

Reply:
right but the whole file will have more dependencies... what I am attempting to explain is that by *only* including `#[tokio::main]` with a program that actually does nothing, I still need to include two features: "macros" and either  ("rt-threaded" or "rt-core")  

Maybe I need text that say something exactly like that.  Hmmm.

Reply:
Sorry, what I wrote above doesn't make sense even to me.

What I meant to say is replace this:

```md
If we change our code, so that it only uses the `#[tokio::main]` macro:
```

with 
```md
If we replace our `src/main.rs` with this (basically hello world w/ with the `#[tokio:main]` macro:
```

Maybe it's clear to others but the way it was written I wasn't sure whether you meant to replace the whole file or just the main function.

Reply:
fixed